pkg/expression/constant_propagation.go

```diff
@@ -472,16 +502,20 @@ var propSpecialJoinConstSolverPool = sync.Pool{
 	},
 }
 
+// PushDownFuncType is to get the type of the join condition
+type PushDownFuncType func(Expression) (eqCond []*ScalarFunction, leftCond []Expression,
```

Comment:
Please add more comments for this function, especially explain why and how to use it.

Reply:
I have refactored this code and added comments for it.